.github/workflows/release.yml

```
@@ -51,24 +53,31 @@ jobs:
         extra_nix_config: |
           experimental-features = nix-command
     - uses: cachix/cachix-action@v10
+      if: startsWith(github.ref, 'refs/heads/')
+      with:
+        name: ic-hs-test
+        # NB: No auth token, we don’t want to push new stuff here
+    - uses: cachix/cachix-action@v10
+      if: startsWith(github.ref, 'refs/tags/')
       with:
         name: ic-hs-test
         authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
     - run: cachix watch-store ic-hs-test &
     - name: "nix-build"
+      env:
+        GC_DONT_GC: 1 # otherwise `nix` crash on macOS
       # these are the dependencies listed in release-files. Sorry for the duplication
       run: |
-        env GC_DONT_GC=1 nix-build --max-jobs 1 --argstr releaseVersion ${{needs.changelog.outputs.version}} \
+        nix-build --max-jobs 1 --argstr releaseVersion ${{needs.changelog.outputs.version}} \
           -A moc -A mo-ide -A mo-doc -A js.moc -A js.moc_interpreter
 
   # Finally do the upload. Hopefully the previous job has uploaded the
-  # build product to the nix cache, as we cannot build the darwin products on
+  # build product to the cachix cache, as we cannot build the darwin products on
   # linux
   release:
+    if: startsWith(github.ref, 'refs/tags/')
     runs-on: 'ubuntu-latest'
-    needs:
-    - changelog
-    - build
+    needs: [ changelog, build ]
```

`build` should be enough?
```suggestion
    needs: build
```

---

nope, it needs the version from the `changelog`. We could use `${{ github.ref_name }}` instead!

---

Actually `needs.changelog.outputs.release_body` is a prerequisite too.